packages/core/core/src/fs.ts

```diff
@@ -0,0 +1,92 @@
+import parseJson from 'parse-json';
+import * as path from 'path';
+import 'colors';
+
+export function mkdirpSync(...args) { return fs.mkdirpSync(...args); }
+
+export function mkdirp(...args) { return fs.mkdirp(...args); }
+
+export function readdir(...args) { return fs.readdir(...args); }
+
+export function stat(...args) { return fs.stat(...args); }
+
+export function remove(...args) { return fs.remove(...args); }
+
+export function copy(...args) { return fs.copy(...args); }
+
+export function copySync(...args) { return fs.copySync(...args); }
+
+export function move(...args) { return fs.move(...args); }
+
+export function moveSync(...args) { return fs.moveSync(...args); }
+
+export function symlink(...args) { return fs.symlink(...args); }
+
+export function appendFileSync(...args) { return fs.appendFileSync(...args); }
+
+export function writeFile(...args) { return fs.writeFile(...args); }
+
+export function writeFileSync(...args) { return fs.writeFileSync(...args); }
+
+export function readFile(...args) { return fs.readFile(...args); }
+
+export function readFileSync(...args) { return fs.readFileSync(...args); }
+
+export function readdirSync(...args) { return fs.readdirSync(...args); }
+
+export function statSync(...args) { return fs.statSync(...args); }
+
+export function readJSONSync(...args) {
+  let json;
+  try {
+    json = parseJson(readFileSync(...args));
+  } catch (e) {
+    console.error('error: '.red + args[0].green.underline + ' ' + e.message.green);
+    process.exit(1);
+  }
+  return json;
+}
+
+export function writeJSONSync(...args) { return fs.writeJSONSync(...args); }
+
+export function outputJSONSync(...args) { return fs.outputJSONSync(...args); }
+
+export function writeJson(...args) { return fs.writeJson(...args); }
+
+export function existsSync(...args) { return fs.existsSync(...args); }
+
+export function ensureFileSync(...args) { return fs.ensureFileSync(...args); }
+
+export function ensureDirSync(...args) { return fs.ensureDirSync(...args); }
+
+export function access(...args) { return fs.access(...args); }
+
+export function removeSync(...args) { return fs.removeSync(...args); }
+
+export function createWriteStream(...args) { return fs.createWriteStream(...args); }
+
+export function copyPreserve(sourceFilePath, targetFilePath) {
+  const implementation = (sourceFilePath, targetFilePath) => {
+    let ext = 1;
+    let preserved = targetFilePath;
+    while (fs.existsSync(preserved)) {
+      const extname = path.extname(targetFilePath);
+      preserved = joinPath(
+        path.dirname(targetFilePath),
+        `${path.basename(targetFilePath, extname)}.${ext}${extname}`
+      );
+      ext++;
+    }
+    if (preserved !== targetFilePath) {
+      fs.copySync(targetFilePath, preserved);
+    }
+    fs.copySync(sourceFilePath, targetFilePath);
+  };
+
+  return implementation(sourceFilePath, targetFilePath);
+}
+
+export function outputFileSync(...args) { return fs.outputFileSync(...args); }
```

Comment:
AFAICS there's only two APIs that are specific to Embark (not coming from `fs-extra`). There good opportunity to get rid of most of what's in this module as most functions used are coming from `fs-extra`.